homeassistant/components/vesync/switch.py

```diff
@@ -1,29 +1,57 @@
 """Support for VeSync switches."""
 
+from collections.abc import Callable
+from dataclasses import dataclass
 import logging
-from typing import Any
+from typing import Any, Final
 
 from pyvesync.vesyncbasedevice import VeSyncBaseDevice
-
-from homeassistant.components.switch import SwitchEntity
+from pyvesync.vesyncoutlet import VeSyncOutlet
+from pyvesync.vesyncswitch import VeSyncSwitch
+
+from homeassistant.components.switch import (
+    SwitchDeviceClass,
+    SwitchEntity,
+    SwitchEntityDescription,
+)
 from homeassistant.config_entries import ConfigEntry
 from homeassistant.core import HomeAssistant, callback
 from homeassistant.helpers.dispatcher import async_dispatcher_connect
 from homeassistant.helpers.entity_platform import AddEntitiesCallback
 
-from .const import DEV_TYPE_TO_HA, DOMAIN, VS_COORDINATOR, VS_DEVICES, VS_DISCOVERY
+from .common import is_outlet, is_wall_switch
+from .const import DOMAIN, VS_COORDINATOR, VS_DEVICES, VS_DISCOVERY
 from .coordinator import VeSyncDataCoordinator
 from .entity import VeSyncBaseEntity
 
 _LOGGER = logging.getLogger(__name__)
 
 
+@dataclass(frozen=True, kw_only=True)
+class VeSyncSwitchEntityDescription(SwitchEntityDescription):
+    """A class that describes custom switch entities."""
+
+    is_on: Callable[[VeSyncBaseDevice], bool]
+    exists_fn: Callable[[VeSyncBaseDevice], bool]
+
+
+SENSOR_DESCRIPTIONS: Final[tuple[VeSyncSwitchEntityDescription, ...]] = (
+    VeSyncSwitchEntityDescription(
+        key="device_status",
+        is_on=lambda device: device.device_status == "on",
+        # Other types of wall switches support dimming.  Those use light.py platform.
+        exists_fn=lambda device: is_wall_switch(device) or is_outlet(device),
```

Comment:
Okay, but how come only these 2 are switches? As in, `is_wall_switch` doesnt sound to incorporate dimming

Reply:
I thought I had handled this different maybe lost in a merge - so here is the challenge.  The old lookup table takes two model of switches and calls them a dimmer switch.  Goal is to not need that lookup table for models in the future. 

https://github.com/cdnninja/core/blob/09ae388f4e947f508cedda70b806a839326ea4af/homeassistant/components/vesync/const.py#L51-L52

These devices in the old method didn't create a switch entity just a light entity.    Currently with my code it would also get a switch.    Should we leave it like that so if dimmable switch or outlet you still get a switch toggle?      This is the easier way, other option is adjust the exists to exclude the type of VeSyncDimmerSwitch which is a type of switch within the library.  

@iprak  Thoughts?

Reply:
I don't think we need to provided a separate switch for dimmer. [LightEntity extends from ToggleEntity](https://github.com/home-assistant/core/blob/2bedb2cadbfd590c9c94b5a93793c83ce286d55e/homeassistant/components/light/__init__.py#L872) and so it provided on/off capability natively.

This is what I see for a dimmer switch
![image](https://github.com/user-attachments/assets/9311d1f5-aa26-4c46-a578-a63e78adbc8e)

Reply:
Went back to fix this and realized I was correct up front.   In pyvesync vesyncswitch has two child classes, VeSyncWallSwitch and VeSyncDimmerSwitch.   The above filters to just wall switch so dimmer isn't used.   As such this won't show for dimmers.   No changes required.